lib/index.js

```diff
@@ -2,7 +2,94 @@ const crypto = require('crypto');
 
 const ALPHA_NUM = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
 
-const IdGenerator = function(prefixes){
+// Largest integer we can read from a buffer.
+//const RAND_MAX = 0xFFFF_FFFF_FFFF;
+const RAND_MAX = Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+/*
+ Generates an integer in [min, max) range where min is inclusive and max is
+ exclusive.
+
+ Backported from node.js v14 crypto.randomInt,
+ https://nodejs.org/api/crypto.html#crypto_crypto_randomint_min_max_callback
+ This version eplaces native NumberIsSafeInteger 
+ with Number.isSafeInteger and uses locally defined rand_max values
+
+ This function is licensed under the terms of the nodejs license,
+ please see the NODEJS_LICENSE file in the root of this repository
+*/
+function randomInt(min, max, callback) {
+  // Detect optional min syntax
+  // randomInt(max)
+  // randomInt(max, callback)
+  const minNotSpecified = typeof max === 'undefined' ||
+    typeof max === 'function';
+
+  if (minNotSpecified) {
+    callback = max;
+    max = min;
+    min = 0;
+  }
+
+  const isSync = typeof callback === 'undefined';
+  if (!isSync && typeof callback !== 'function') {
+    throw new ERR_INVALID_CALLBACK(callback);
+  }
+  if (!Number.isSafeInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'a safe integer', min);
+  }
+  if (!Number.isSafeInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'a safe integer', max);
+  }
+  if (max <= min) {
+    throw new ERR_OUT_OF_RANGE(
+      'max', `greater than the value of "min" (${min})`, max
+    );
+  }
+
+  // First we generate a random int between [0..range)
+  const range = max - min;
+
+  if (!(range <= RAND_MAX)) {
+    throw new ERR_OUT_OF_RANGE(`max${minNotSpecified ? '' : ' - min'}`,
+      `<= ${RAND_MAX}`, range);
+  }
+
+  const excess = RAND_MAX % range;
+  const randLimit = RAND_MAX - excess;
+
+  if (isSync) {
+    // Sync API
+    while (true) {
+      const x = crypto.randomBytes(6).readUIntBE(0, 6);
+      // If x > (maxVal - (maxVal % range)), we will get "modulo bias"
+      if (x > randLimit) {
+        // Try again
+        continue;
+      }
+      const n = (x % range) + min;
+      return n;
+    }
+  } else {
+    // Async API
```

Comment:
Not a big deal since this is well-tested code that is brought over: my concern is adding code that may not be exercised in tests nor in practice since we don't expose an async API for this lib and we may deviate in the sync implementation over time.

Reply:
I've removed the async code path from the function now, and created a set of unit tests that hopefully give some good coverage.